src/extensions/step_handlers/swupdate_handler_v2/src/swupdate_handler_v2.cpp

```diff
@@ -262,12 +262,36 @@ ADUC_Result SWUpdateHandler_PerformAction(
         workflow_peek_result_details(workflowData->WorkflowHandle));
 
 done:
-    if (IsAducResultCodeFailure(result.ResultCode) && workflowData->WorkflowHandle != nullptr)
+    if (IsAducResultCodeFailure(result.ResultCode))
     {
         workflow_set_result(workflowData->WorkflowHandle, result);
         workflow_set_state(workflowData->WorkflowHandle, ADUCITF_State_Failed);
     }
 
+    // Note: the handler must request a system reboot or agent restart if required.
```

Comment:
this method is already  pretty big, and these results only apply to install and result.  should n't these be checked in the install and apply handlers specifically?

also, when did we add reboot after install?  I believe it was originally only for apply which makes more sense to me.  even if a handler doesn't have work to do in apply , it can still return reboot from that,

Reply:
interestingly, it looks like this used to be done in apply like i suggest, but you've moved it here?

Reply:
Customer can choose to request the agent to reboot or restart even during install phase. IMO, in the future, we need to decouple any workflow-related request from the resultCode, and provide the APIs for the handlers, or backing scripts to make a request to the agent core directly.

As for this change, I'm making this behavior consistent with the Script Handler.

Reply:
(FWIW, we can even completely deprecate SWUpdate Handler and SWUpdate handler V2, since the same update scenario can be accomplished using Script Handler)

Reply:
your replies are related to the second half of my comments.  My first comment still applies however:

*this method is already pretty big, and these results only apply to install and result. shouldn't these be checked in the install and apply handlers specifically?*

Reply:
Fixed